core/lib/protobuf_config/src/da_client.rs

```diff
@@ -17,17 +17,36 @@ impl ProtoRepr for proto::DataAvailabilityClient {
         let config = required(&self.config).context("config")?;
 
         let client = match config {
-            proto::data_availability_client::Config::Avail(conf) => Avail(AvailConfig {
-                api_node_url: required(&conf.api_node_url)
-                    .context("api_node_url")?
-                    .clone(),
-                bridge_api_url: required(&conf.bridge_api_url)
-                    .context("bridge_api_url")?
-                    .clone(),
-                app_id: *required(&conf.app_id).context("app_id")?,
-                timeout: *required(&conf.timeout).context("timeout")? as usize,
-                max_retries: *required(&conf.max_retries).context("max_retries")? as usize,
-            }),
+            proto::data_availability_client::Config::Avail(conf) => {
+                Avail(match conf.config.as_ref() {
```

Comment:
nit: you can move parsing shared params (`bridge_api_url`, `timeout`) out of this match

Reply:
Fixed, closing.